src/GlobalPayments.Api/Terminals/HPA/Responses/SipBaseResponse.cs

```diff
@@ -256,5 +256,7 @@ public class SipTerminalResponse : SipBaseResponse, ITerminalResponse {
 
     public class SipTerminalReport : SipBaseResponse, ITerminalReport {
         internal SipTerminalReport(byte[] buffer, params string[] messageIds) : base(buffer, messageIds) { }
+
+        public string ReportData { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
```

Comment:
This should just be get; set; and returning null instead of throwing an exception if not set here.